src/odemis/driver/hamamatsurx.py

```diff
@@ -773,50 +938,123 @@ def _setTimeRangeFactor(self, value):
+        metadata.
+        """
+        super().__init__(name, role, parent=parent, daemon=daemon, **kwargs)
 
-        self.parent = parent
+        self._streak_unit = streak_unit
         self.location = "Delaybox"  # don't change, internally needed by HPDTA/RemoteEx
 
         self._hwVersion = parent.DevParamGet(self.location, "DeviceName")[0]   # needs to be a string
         self._metadata[model.MD_HW_VERSION] = self._hwVersion
 
+        avail_params = parent.DevParamsList(self.location)
+
+        # FIXME: on synchroscan unit, the parameters are quite different.
+        # On synchroscan, DevParamsList: 'Delay Time', 'Lock Mode', 'Device Status'
         # Set parameters delay generator
-        parent.DevParamSet(self.location, "Setting", "M1")  # TODO might be enough and don't need the rest...check!!
-        parent.DevParamSet(self.location, "Trig. Mode", "Ext. rising")  # Note: set to "Int." for testing without SEM
-        parent.DevParamSet(self.location, "Repetition Rate", 1000000)  # [0.001, 10000000] # read-only for Ext. rising
-        parent.DevParamSet(self.location, "Delay A", 0)
-        parent.DevParamSet(self.location, "Delay B", 0.00000002)
-        parent.DevParamSet(self.location, "Burst Mode", "Off")
-
-        self._metadata[model.MD_TRIGGER_DELAY] = self.GetTriggerDelay()
+        if "Setting" in avail_params:
+            parent.DevParamSet(self.location, "Setting", "M1")  # TODO might be enough and don't need the rest...check!!
+        if "Trig. Mode" in avail_params:
+            parent.DevParamSet(self.location, "Trig. Mode", "Ext. rising")  # Note: set to "Int." for testing without SEM
+        # Note: this is legacy code, to maintain the behavior of the old version
+        # A better way would be to use the "properties" in the microscope file, to set the triggerDelay (Delay A)
+        # and delayB values just after initialisation.
+        if "Delay A" in avail_params:
+            parent.DevParamSet(self.location, "Delay A", 0)
+        if "Delay B" in avail_params:
+            parent.DevParamSet(self.location, "Delay B", 0.00000002)
+        if "Burst Mode" in avail_params:
+            parent.DevParamSet(self.location, "Burst Mode", "Off")
+
         # Note: trigger rate (repetition rate) corresponds to the ebeam blanking frequency (read only in RemoteEx)
-        self._metadata[model.MD_TRIGGER_RATE] = int(self.parent.DevParamGet(self.location, "Repetition Rate")[0])
+        if "Repetition Rate" in avail_params:
+            self.triggerRate = model.FloatVA(0, unit="Hz", readonly=True, getter=self.GetTriggerRate)
 
-        triggerDelay = self.GetTriggerDelay()
-        range_trigDelay = self._getTriggerDelayRange()
-        self.triggerDelay = model.FloatContinuous(triggerDelay, range_trigDelay, setter=self._setTriggerDelay, unit="s")
+        # VAs
+        self._delay_setters: Dict[str, callable] = {}  # name of the RemoteEx param -> setter function
```

Comment:
```suggestion
        self._delay_setters: Dict[str, Callable] = {}  # name of the RemoteEx param -> setter function
```
Callable should be from typing